### ISSUE TITLE
Default formatter for a language not set when using devcontainer

### ISSUE THREAD

**Issue by rectalogic** [microsoft/vscode#175152]
I'm using this extension in devcontainers, below is my `.devcontainer/devcontainer.json`.
It does not format on save, and if I choose `Format Document` from the menu on a python file it displays a dialog `Configure Default Formatter` and then prompts me to choose `ms-python.black-formatter` or `ms-python.python` (I assume this is why format on save doesn't work, it does sees two formatters and for some reason doesn't know the default one).

So it seems to be ignoring both my `"editor.defaultFormatter": "ms-python.black-formatter"` and `"python.formatting.provider": "none"` settings.

How can I debug this further?

```json
{
    "name": "Cureatr Ops",
    "image": "cureatr/cureatr-vm",
    "init": true,
    "remoteUser": "cureatr",
    "overrideCommand": false,
    "remoteEnv": {
        "PATH": "/site/venv/bin:${containerEnv:PATH}",
        "PYTHONPATH": "${containerWorkspaceFolder}/ec2"
    },
    "customizations": {
        "vscode":{
            "extensions": [
                "ms-python.python",
                "ms-python.black-formatter",
                "charliermarsh.ruff",
                "eamodio.gitlens"
            ],
            "settings": {
                "telemetry.telemetryLevel": "off",
                "python.defaultInterpreterPath": "/site/venv/bin/python",
                "python.languageServer": "Pylance",
                "python.linting.lintOnSave": true,
                "python.linting.pylintEnabled": false,
                "ruff.interpreter": ["/site/venv/bin/python"],
                "ruff.importStrategy": "fromEnvironment",
                "ruff.organizeImports": true,
                "ruff.fixAll": false,
                "python.formatting.provider": "none",
                "black-formatter.importStrategy": "fromEnvironment",
                "source.organizeImports": true,
                "git.ignoreLimitWarning": true,
                "[python]": {
                    "editor.defaultFormatter": "ms-python.black-formatter",
                    "editor.formatOnSave": true,
                    "editor.codeActionsOnSave": [
                        "source.organizeImports"
                    ]
                }
            }
        }
    }
}
```

**Comment by rectalogic** [comment]
When I pick a formatter when prompted, the setting seems to be going nowhere, I haven't found it and if I choose to format multiple times in a row, it re-prompts me every time that there is no default formatter and to choose one.

**Comment by karthiknadig** [comment]
Moved this to VS Code for further investigation. This seems to be a bug with how the default formatter configuration is handled.

**Comment by rectalogic** [comment]
I setup a minimal reproduction here https://github.com/rectalogic/vscode-formatter

Open that project in vscode and if you edit the included python file, it does not reformat on save. And if you choose `Format Document` it prompts to choose a default formatter.

**Comment by rectalogic** [comment]
Seems to be fixed as of vscode 1.76.0
